meilidb-http/src/routes/index.rs

```diff
@@ -12,104 +17,312 @@ use crate::models::token::ACL::*;
 use crate::routes::document::IndexUpdateResponse;
 use crate::Data;
 
+fn generate_uid() -> String {
+    let mut rng = rand::thread_rng();
+    let sample = b"abcdefghijklmnopqrstuvwxyz0123456789";
+    sample
+        .choose_multiple(&mut rng, 8)
+        .map(|c| *c as char)
+        .collect()
+}
+
 pub async fn list_indexes(ctx: Context<Data>) -> SResult<Response> {
     ctx.is_allowed(IndexesRead)?;
-    let list = ctx
-        .state()
-        .db
-        .indexes_names()
-        .map_err(ResponseError::internal)?;
-    Ok(tide::response::json(list))
+
+    let indexes_uids = ctx.state().db.indexes_uids();
+
+    let env = &ctx.state().db.env;
+    let mut reader = env.read_txn().map_err(ResponseError::internal)?;
+
+    let mut response_body = Vec::new();
+
+    for index_uid in indexes_uids {
+        let index = ctx
+            .state()
+            .db
+            .open_index(&index_uid)
+            .ok_or(ResponseError::internal(&index_uid))?;
+        let name = index
+            .main
+            .name(&mut reader)
+            .map_err(ResponseError::internal)?
+            .ok_or(ResponseError::internal("Name not found"))?;
+        let created_at = index
+            .main
+            .created_at(&mut reader)
+            .map_err(ResponseError::internal)?
+            .ok_or(ResponseError::internal("Created date not found"))?;
+        let updated_at = index
+            .main
+            .updated_at(&mut reader)
+            .map_err(ResponseError::internal)?
+            .ok_or(ResponseError::internal("Updated date not found"))?;
+
+        let index_reponse = IndexResponse {
+            name,
+            uid: index_uid,
+            created_at,
+            updated_at,
+        };
+        response_body.push(index_reponse);
+    }
+
+    Ok(tide::response::json(response_body))
 }
 
-pub async fn get_index_schema(ctx: Context<Data>) -> SResult<Response> {
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+struct IndexResponse {
+    name: String,
+    uid: String,
+    created_at: DateTime<Utc>,
+    updated_at: DateTime<Utc>,
+}
+
+pub async fn get_index(ctx: Context<Data>) -> SResult<Response> {
     ctx.is_allowed(IndexesRead)?;
 
     let index = ctx.index()?;
 
     let env = &ctx.state().db.env;
-    let reader = env.read_txn().map_err(ResponseError::internal)?;
+    let mut reader = env.read_txn().map_err(ResponseError::internal)?;
 
-    let schema = index
+    let uid = ctx.url_param("index")?.to_string();
+    let name = index
         .main
-        .schema(&reader)
-        .map_err(ResponseError::create_index)?;
+        .name(&mut reader)
```

Comment:
Why does this name function need a mutable reader? Asking for a mutable reader is useless.